examples/ex2/main.cpp

```diff
@@ -0,0 +1,271 @@
+#include "MCIntegrator.hpp"
+#include "MCIObservableFunctionInterface.hpp"
+#include "FeedForwardNeuralNetwork.hpp"
+
+#include <iostream>
+
+/*
+Build an example for MCI++:
+
+compute the integral
+
+         \int_{-10}^{+10} dx x^2 nn(x)^2
+(nn(x) is a normalized neural network) with MC and a non-MC method
+
+do the same with
+
+         \int_{-10}^{+10} dx \laplacian^2 nn(x)^2
+*/
+
+
+
+
+
+class NN2: public MCIObservableFunctionInterface{
+private:
+    FeedForwardNeuralNetwork * _ffnn;
+
+public:
+    NN2(FeedForwardNeuralNetwork * ffnn): MCIObservableFunctionInterface(1, 1){
+        _ffnn = ffnn;
+    }
+
+    virtual void observableFunction(const double * in, double *out){
+        _ffnn->setInput(1, in);
+        _ffnn->FFPropagate();
+        const double v = _ffnn->getOutput(1);
+        out[0] = v*v;
+    }
+
+};
+
+
+
+class NN2X2: public MCIObservableFunctionInterface{
+private:
+    FeedForwardNeuralNetwork * _ffnn;
+
+public:
+    NN2X2(FeedForwardNeuralNetwork * ffnn): MCIObservableFunctionInterface(1, 1){
+        _ffnn = ffnn;
+    }
+
+    virtual void observableFunction(const double * in, double *out){
+        _ffnn->setInput(1, in);
+        _ffnn->FFPropagate();
+        const double v = _ffnn->getOutput(1);
+        out[0] = v*v*in[0]*in[0];
+    }
+
+};
+
+
+class X2: public MCIObservableFunctionInterface{
+private:
+    FeedForwardNeuralNetwork * _ffnn;
+
+public:
+    X2(): MCIObservableFunctionInterface(1, 1){}
+
+    virtual void observableFunction(const double * in, double *out){
+        out[0] = in[0]*in[0];
+    }
+
+};
+
+
+
+
+class NN2Sampling: public MCISamplingFunctionInterface{
+private:
+    FeedForwardNeuralNetwork * _ffnn;
+public:
+    NN2Sampling(FeedForwardNeuralNetwork * ffnn): MCISamplingFunctionInterface(1, 1){
+        _ffnn = ffnn;
+    }
+
+    void samplingFunction(const double *in, double * protovalues){
+        _ffnn->setInput(1, in);
+        _ffnn->FFPropagate();
+        const double v = _ffnn->getOutput(1);
+        protovalues[0] = v*v;
+    }
+
+    double getAcceptance(){
+        return getProtoNew(0)/getProtoOld(0);
+    }
+};
+
+
+class NN2Lapl2: public MCIObservableFunctionInterface{
+private:
+    FeedForwardNeuralNetwork * _ffnn;
+
+public:
+    NN2Lapl2(FeedForwardNeuralNetwork * ffnn): MCIObservableFunctionInterface(1, 1){
+        _ffnn = ffnn;
+    }
+
+    virtual void observableFunction(const double * in, double *out){
+        _ffnn->setInput(1, in);
+        _ffnn->FFPropagate();
+        const double v = _ffnn->getOutput(1);
+        const double d2 = _ffnn->getSecondDerivative(1, 0);
+        out[0] = v*v*d2;
```

Comment:
Maybe im confused, but to me it seems this is int nn(x)^2 lapl(nn(x)) dx , instead of int lapl^2 nn(x)^2 as you say above. By the way, what is lapl^2? Laplacian is already second deriv, not? Why the ^2 then?

Reply:
yeah sorry, I was confusing!
I should have written nabla^2 or laplacian.
I'll change that